pandas/tests/indexing/test_categorical.py

```diff
@@ -533,3 +533,19 @@ def test_loc_with_non_string_categories(self, idx_values, ordered):
         result.loc[sl, "A"] = ["qux", "qux2"]
         expected = DataFrame({"A": ["qux", "qux2", "baz"]}, index=cat_idx)
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "cat_idx",
+        [
+            # No duplicates
+            CategoricalIndex(["A", "B"]),
+            # Duplicates: GH#38906
+            CategoricalIndex(["A", "A"]),
+        ],
+    )
+    def test_reindex_empty(self, cat_idx):
```

Comment:
can you move to pandas/tests/frame//indexing/test_indexing.py

also add the cases with Index from the OP which already work